src/gausskernel/storage/access/k2/catalog/sql_catalog_client.cpp

```diff
@@ -188,5 +189,40 @@ sh::Status SqlCatalogClient::IncrementCatalogVersion() {
   K2LOG_D(log::catalog, "IncrementCatalogVersion took {}", k2::Clock::now() - start);
   return std::get<0>(result);
 }
+
+sh::Status SqlCatalogClient::GetAttrNumToSKVOffset(uint32_t database_oid, uint32_t relation_oid, std::unordered_map<int, uint32_t>& attr_to_offset) {
+    auto [status, tableInfo] = catalog_manager_->GetTableSchema(database_oid, relation_oid);
+    if (!status.is2xxOK()) {
+        return status;
+    }
+    PgTableDesc desc(tableInfo);
+    auto colmap = desc.GetAttrNumToColMap();
+    for (const auto& [attr_num, col] : colmap) {
+        attr_to_offset[attr_num] = col->index();
```

Comment:
I think we should do "col->index() + 2" so that this includes the two built in fields we use on SKV for the table id and index id.

Reply:
Done